src/Motor.Extensions.Hosting.Kafka/KafkaMessageConsumer.cs

```diff
@@ -162,47 +166,43 @@ private void WriteStatistics(string json)
         }
     }
 
-    private void SingleMessageHandling(CancellationToken token, ConsumeResult<string?, byte[]> msg)
+    private async Task<bool> SingleMessageHandlingAsync(CancellationToken token, ConsumeResult<string?, byte[]> msg)
     {
-        _logger.LogDebug(
-            $"Received message from topic '{msg.Topic}:{msg.Partition}' with offset: '{msg.Offset}[{msg.TopicPartitionOffset}]'");
+        _logger.LogDebug("Received message from topic '{Topic}:{Partition}' with offset: '{Offset}[{TopicPartitionOffset}]'",
+            msg.Topic, msg.Partition, msg.Offset, msg.TopicPartitionOffset);
         var cloudEvent = KafkaMessageToCloudEvent(msg.Message);
 
-        var taskAwaiter = ConsumeCallbackAsync?.Invoke(cloudEvent, token).GetAwaiter();
-        taskAwaiter?.OnCompleted(() =>
+        return await await ConsumeCallbackAsync?.Invoke(cloudEvent, token)
```

Comment:
Also, wouldn't it be better to lighten this a bit up by making this two separate calls? Like this:
```c#
        var status = await ConsumeCallbackAsync!.Invoke(cloudEvent, token);
        return HandleMessageStatus(msg, status);
```

Reply:
>  would block on status

What do you mean by that? With @pkemkes's suggestion, we would `await` the `status`. How is that blocking?

Reply:
Yes, I was wrong here. See https://github.com/GDATASoftwareAG/motornet/pull/747#discussion_r1095620278